activejob/lib/active_job/test_helper.rb

```diff
@@ -323,11 +332,11 @@ def clear_performed_jobs
         performed_jobs.clear
       end
 
-      def enqueued_jobs_size(only: nil)
-        if only
-          enqueued_jobs.count { |job| Array(only).include?(job.fetch(:job)) }
-        else
-          enqueued_jobs.count
+      def enqueued_jobs_size(only: nil, queue: nil)
+        enqueued_jobs.count do |job|
+          job_class = job.fetch(:job)
+          (only.nil? || Array(only).include?(job_class)) &&
+          (queue.nil? || job.fetch(:queue, job_class.queue_name) == queue)
```

Comment:
I find this new code to be rather confusing to read. Can you think of a better way to write it? Even if there are more lines I'd like it to be more readable.